site/content/blog/kapp_deploy_oidc_gke.md

```diff
@@ -0,0 +1,119 @@
+
+- No cloud secrets: No need to duplicate cloud credentials as long-lived GitHub secrets. Instead, you can configure the OIDC trust on GCP, and then update workflows to request a short-lived access token from the cloud provider through OIDC.
+- Authentication and authorization management: Will have more granular control over how workflows can use credentials and also you can control access to cloud resources.
+- Rotating credentials: With OIDC, cloud provider issues a short-lived access token that is only valid for a single job, and then automatically expires.
+
+### How
+
+Now we will see how we can use GitHub Action – [auth](https://github.com/google-github-actions/auth) to set up and configure authentication to Google Cloud. We need to perform the following configurations on GCP - 
+
+1. Create a new Workload Identity Pool (IAM -> Workload Identity Federation -> Workload Identity Pool) and add an OIDC Provider to it with Issuer URL as `https://token.actions.githubusercontent.com`.
+2. Configure the Attribute mapping and conditions of the provider.
+3. Create a service account and connect Workload Identity Pool you just created to the service account by assigned to the Workload Identity User role. For more information, see the [GCP documentation](https://cloud.google.com/iam/docs/workload-identity-federation?_ga=2.114275588.-285296507.1634918453#conditions).
+
+To update workflows for OIDC, you will need to make two changes to your YAML:
+
+1. Add permissions settings for the token. The job or workflow run requires a permissions setting with `id-token: write`. You won't be able to request the OIDC JWT ID token if the permissions setting for `id-token` is set to `read` or `none`.
+
+```yaml
+permissions:
+  id-token: write
+```
+
+2. Use the google-github-actions/auth action to exchange the OIDC token (JWT) for a cloud access token.
+
+```yaml
+steps:
+- id: 'auth'
+  name: 'Authenticate to Google Cloud'
+  uses: 'google-github-actions/auth@v0.4.0'
+  with:
+    workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+    service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+```
+
+This will use the configured workload_identity_provider and service_account to authenticate future steps. Make sure to replace the value of `workload_identity_provider` with the path to your identity provider in GCP and also replace the value of `service_account` key with the name of your service account in GCP. 
+
+### Example
+
+Here is a sample Github action which gets triggered when a new tag is created on the repo. It authenticates with GCP, gets the GKE credentials, install carvel tools on the GKE cluster and deploy a simple app using kapp. 
+
+```yaml
+name: OIDC auth GCP
+
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  Get_OIDC_ID_token:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: 'write'
+      contents: 'read'
```

Comment:
Above you have mentioned adding only `id-token` to permissions, but now there are `contents: read` as well. If it is not very useful, maybe we can remove it.

Reply:
Yeah we can remove `contents: 'read'`. Done.